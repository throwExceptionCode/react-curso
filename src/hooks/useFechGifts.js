import { useState,useEffect } from "react";
import { getGif } from '../helpers/getGifs';

export const useFetchGifs = ( category )=>
{
    const [state, setstate] = useState({
        data:[],
        loading:true,
    });
    
    
    useEffect(()=>{
        getGif(category).then(imgs => setstate({
            data:imgs,
            loading:false,
        }));
    },[category])

    return state;
}