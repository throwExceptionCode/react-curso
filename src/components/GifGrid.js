import React from 'react';//, { useState,useEffect }
import { useFetchGifs } from '../hooks/useFechGifts';
//import { getGif } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

const GifGrid = ({ category }) => {
    //const [Images, setImages] = useState([]);
    const { data:images,loading } = useFetchGifs(category);
    //const [conteo, setconteo] = useState(1);
    /*useEffect(()=>{
        getGif(category).then(setImages);
    },[category])*/
    //<h2>{conteo}</h2>
    //<button onClick={ () => setconteo( conteo + 1 ) } >+</button>
    
    
    
    //getGif

    useFetchGifs();
    
    return (
        <>
            <li>
                <div className="card-grid" >
                    <h3  className="title-search" > { category } </h3>
                    {loading && 'Cargando' }
                    <br/>
                    { 
                        images.map( (img) => (
                            <GifGridItem key={img.id}  { ...img } />
                        ) )
                    }
                </div>
            </li>

                
        </>
    )
}

export default GifGrid
