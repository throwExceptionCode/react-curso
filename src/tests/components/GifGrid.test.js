import React from 'react';
import { shallow } from "enzyme";
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFechGifts';

jest.mock('../../hooks/useFechGifts');

describe('Pruebas con GifGrid', () => {
    
    
    test('should ber match with snapshot', () => {
        const category = 'got';
        //falsear data
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should ber show img items useFetch', () => {
        //falsear data
        const gifs = [
            {
                id: 1, 
                title: 'img.title', 
                url: 'img.images.downsized_medium.url'
            },
            {
                id: 2, 
                title: 'img.title', 
                url: 'img.images.downsized_medium.url'
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const category2 = 'got2';
        const wrapper = shallow(<GifGrid category={category2} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
