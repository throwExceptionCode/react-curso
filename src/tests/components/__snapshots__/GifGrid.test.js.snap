// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGrid should ber match with snapshot 1`] = `
<Fragment>
  <li>
    <div
      className="card-grid"
    >
      <h3
        className="title-search"
      >
         
        got
         
      </h3>
      Cargando
      <br />
    </div>
  </li>
</Fragment>
`;

exports[`Pruebas con GifGrid should ber show img items useFetch 1`] = `
<Fragment>
  <li>
    <div
      className="card-grid"
    >
      <h3
        className="title-search"
      >
         
        got2
         
      </h3>
      <br />
      <GifGridItem
        id={1}
        key="1"
        title="img.title"
        url="img.images.downsized_medium.url"
      />
      <GifGridItem
        id={2}
        key="2"
        title="img.title"
        url="img.images.downsized_medium.url"
      />
    </div>
  </li>
</Fragment>
`;
